modules/reindex/src/main/java/org/elasticsearch/reindex/remote/RemoteRequestBuilders.java

```diff
@@ -140,11 +141,22 @@ static Request initialSearch(SearchRequest searchRequest, BytesReference query,
                 }
             }
 
-            if (searchRequest.source().fetchSource() != null) {
-                entity.field("_source", searchRequest.source().fetchSource());
-            } else {
-                if (remoteVersion.onOrAfter(Version.fromId(1000099))) {
-                    // Versions before 1.0 don't support `"_source": true` so we have to ask for the source as a stored field.
+            if (remoteVersion.onOrAfter(Version.fromId(1000099))) {
+                // Versions before 1.0 don't support `_source` so we have to ask for the source as a stored field.
+                if (searchRequest.source().fetchSource() != null) {
+                    FetchSourceContext fetchSource = searchRequest.source().fetchSource();
+                    if (remoteVersion.before(Version.V_8_16_0)) {
```

Comment:
Once I found this file, it looked much simpler to put the custom logic for excluding `include_vectors` here instead of building it into `FetchSourceContext`. If you look at the logic in the rest of this method, its entire purpose is to modify the request to be compatible with old ES versions, which is exactly what we need to do to `FetchSourceContext` here.

Reply:
Also, adding the extra boolean does not fix the root cause of the problem. The root cause of the problem is that all communications for a remote reindexing request are through XContent. Additionally, a remote reindex request does not use a special search request to get docs from the remote source, it uses a standard search request. Thus, if we want to keep `semantic_text` vector fields when reindexing from an 8.16+ remote cluster, we must indicate so through the `include_vectors` param, otherwise they will be excluded. This leads to the solution I implemented, where we exclude the `include_vectors` param when communicating with a pre-8.16 cluster, but include it otherwise to maintain desired behavior.

Adding another boolean would not help because it would ultimately need to be added to XContent to be sent to the remote cluster, leading to the same problem.